drivers/display/display_st7789h2.c

```diff
@@ -0,0 +1,645 @@
+	return 0;
+}
+
+static int st7789h2_read(const struct device *dev,
+			const uint16_t x,
+			const uint16_t y,
+			const struct display_buffer_descriptor *desc,
+			void *buf)
+{
+	return -ENOTSUP;
+}
+
+static void st7789h2_set_mem_area(const struct device *dev, const uint16_t x,
+				 const uint16_t y, const uint16_t w, const uint16_t h)
+{
+	uint16_t tmp_data[2];
+
+	struct st7789h2_data *data = (struct st7789h2_data *)dev->data;
+
+	uint16_t ram_x = x + data->x_offset;
+	uint16_t ram_y = y + data->y_offset;
+
+	tmp_data[0] = sys_cpu_to_be16(ram_x);
+	tmp_data[1] = sys_cpu_to_be16(ram_x + w - 1);
+	st7789h2_transmit(dev, ST7789V_CMD_CASET, (uint8_t *)&tmp_data[0], 4);
+
+	tmp_data[0] = sys_cpu_to_be16(ram_y);
+	tmp_data[1] = sys_cpu_to_be16(ram_y + h - 1);
+	st7789h2_transmit(dev, ST7789V_CMD_RASET, (uint8_t *)&tmp_data[0], 4);
+}
+
+static int st7789h2_write(const struct device *dev,
+			 const uint16_t x,
+			 const uint16_t y,
+			 const struct display_buffer_descriptor *desc,
+			 const void *buf)
+{
+	const struct st7789h2_config *config = (const struct st7789h2_config *)dev->config;
+
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT((desc->pitch * ST7789H2_PIXEL_SIZE * desc->height) <= desc->buf_size,
+			"Input buffer to small");
+
+	LOG_DBG("Writing %dx%d (w,h) @ %dx%d (x,y) p=%d n=%d",
+			desc->width, desc->height, x, y, desc->pitch, desc->buf_size);
+
+	st7789h2_set_mem_area(dev, x, y, desc->width, desc->height);
+
+	st7789h2_transmit(dev, ST7789V_CMD_RAMWR, NULL, 0);
+
+	st7789h2_set_cs(config, 1);
+	st7789h2_set_cmd(config, 0);
+
+	uint16_t *buffer = (void *)buf;
+	size_t len = desc->buf_size/ST7789H2_PIXEL_SIZE;
+
+	for (size_t i = 0; i < len; i++) {
+		st7789h2_data_bus_set(dev, buffer[i]);
+		st7789h2_wrn_pulse(config);
+	}
+
+	st7789h2_set_cs(config, 0);
+
+	return 0;
+}
+
+static void *st7789h2_get_framebuffer(const struct device *dev)
+{
+	return NULL;
+}
+
+static int st7789h2_set_brightness(const struct device *dev,
+			   const uint8_t brightness)
+{
+	return -ENOTSUP;
+}
+
+static int st7789h2_set_contrast(const struct device *dev,
+			 const uint8_t contrast)
+{
+	return -ENOTSUP;
+}
+
+static void st7789h2_get_capabilities(const struct device *dev,
+			      struct display_capabilities *capabilities)
+{
+	struct st7789h2_data *data = (struct st7789h2_data *)dev->data;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = data->width;
+	capabilities->y_resolution = data->height;
+
+	capabilities->supported_pixel_formats = PIXEL_FORMAT_RGB_565;
+	capabilities->current_pixel_format = PIXEL_FORMAT_RGB_565;
+
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+}
+
+static int st7789h2_set_pixel_format(const struct device *dev,
+			     const enum display_pixel_format pixel_format)
+{
+
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		return 0;
+	}
+	LOG_ERR("Pixel format change not implemented");
+	return -ENOTSUP;
```

Comment:
```c
	if (pixel_format == PIXEL_FORMAT_RGB_565) {
		return 0;
	}

	LOG_WRN("Pixel format not supported");

	return -ENOTSUP;
```

Reply:
Alright, didn't expect that to be a warn since I returned a ENOTSUP error, but if that's the recommended LOG_WRN for this purpose I'll include it.

Is ERR reserved for fatal blocking errors ?

Reply:
LOG_WRN here is probably arguable. It is legitimate for a subsystem to try out what is supported, error in this case can be confusing for the user. I do not insist on LOG_WRN.

Reply:
Done